src/Service/Messenger.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Service;
+
+
+use App\Model\Message;
+
+class Messenger
+{
+    /**
+     * @var SenderInterface[]
+     */
+    protected $senders = [];
+
+    /**
+     * Messenger constructor.
+     * @param SenderInterface[] $senders
+     */
+    public function __construct(array $senders = [])
```

Comment:
Need better implementation, we can use $message in constructor, to set up $this->sender from $message->getType...